test/samples/ts-satisfies-expression/input.ts

```diff
@@ -0,0 +1,18 @@
+type A = string;
+const b: number = 42 satisfies A;
+
+type C = {
+	name: string;
+};
+
+type D = {
+	name: string;
+	description: string;
+};
+
+const e: D = {
+	name: 'foo',
+	desciption: 'bar'
+};
+
+const f = e satisfies D;
```

Comment:
I'd suggest also having a test which requires parentheses to test the ordering in `expressions_precedence`

I just did a quick search online and this was the first example that popped up:
```
function f() {
  const type = 'foo';
  if (type === 'foo') {
    return true
  }
  (type) satisfies never;
  return false;
}
```

Reply:
Agree, but that's not a good test. The parentheses around `type` are only necessary here, because it's a reserved keyword. Whoever names their variable `type` in a **Type**Script context won't have much luck anyway. I found a better test and added it for `satisfies` and `as`. I'm not sure what the other precendences should indicate and in some cases they don't even make sense. 

I have ordered them nevertheless with my best guesses.